### ISSUE TITLE
Code for VQAv2 evaluation

### ISSUE THREAD

**Issue by vateye** [HaozheZhao/MIC#7]
Hi, is there any code for VQAv2 evaluation?

**Comment by vateye** [comment]
Meanwhile, it is the performance on test-dev about VQA-score?

**Comment by HaozheZhao** [comment]
Hello, we use the VQAv2 evaluation code as the original VQA dataset. Here is the VQA tool provided in the [GitHub](https://github.com/GT-Vision-Lab/VQA/blob/master/PythonEvaluationTools/vqaEvaluation/vqaEval.py).
I believe others also integrate this tool into their code, such as [lavis](https://github.com/salesforce/LAVIS/blob/e4040b13d6120062829ee9625f016f3cd3dd16e6/lavis/common/vqa_tools/vqa_eval.py#L18). You can observe that they also use the code from the VQA tool.

**Comment by vateye** [comment]
It seems that the accuracy did not computed in that way. As suggested here.

https://github.com/HaozheZhao/MIC/blob/77afee26c0854cfd5fafbba83f14d55155910256/tasks/vqa/dataset.py#L406C25-L406C25

Any reproduction results? Besides, I am wondering whether validation performance or test-dev performance is reported.

**Comment by HaozheZhao** [comment]
Furthermore, as mentioned in the paper, we utilized the training set of VQAv2 as the foundation for constructing the MIC dataset. Subsequently, we evaluated and documented the performance based on the karpathy split of the test set.

**Comment by vateye** [comment]
You mean the test-dev right?

**Comment by HaozheZhao** [comment]
Regarding your first question, the accuracy you mentioned is used to evaluate our results during pre-training. It can be seen as an EM score. We utilize it along with other metrics such as loss, Rouge, and Bleu scores to assess the progress of pre-training. Therefore, it serves as the metric we report for our results. So it is actually the mertic that we used to report the result.
As for you second question, you are right that we report the test-dev of the VQAv2 dataset.

**Comment by vateye** [comment]
Thanks for your quick response. As far as I know, InstructBLIP did not include the results of VQAv2. Did you reproduce the results by yourselves by directly applying the official weight?

**Comment by HaozheZhao** [comment]
Yeah, we reproduced all the results by applying their official weights. We have noticed that the official weight of InstructBLIP may not perform as well as claimed in their paper. Maybe due to the difference of prompt setting? 
Anyway, we have replicated all the results presented in our paper, and our model also shares the same settings as the other baselines.

**Comment by HaozheZhao** [comment]
We intend to do so, but due to time constraints and a severe shortage of manpower, we currently do not have the time to tackle this. I strongly believe that the project management is in a chaotic state  : (
Perhaps in the coming weeks, we will gradually test those VLM with the llama-base backbone and release our own. Additionally, we were able to conduct a review of the ongoing project and test whether the decrease in InstructBLIP is a result of the prompt configuration, as nowadays all VLMs appear to be prompt sensitive.

**Comment by HaozheZhao** [comment]
Sadly, I am about to take a rest. You can add a star to the repo and leave your further questions. I will reply ASAP.
